libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -206,6 +206,27 @@ bool OverrideProxy::isModifier() const
 	return holds_alternative<ModifierDefinition const*>(m_item);
 }
 
+bool OverrideProxy::isSameSignature(OverrideProxy const& other) const
```

Comment:
Why don't you use the overrideComparator here?

Reply:
Because for modifiers, the `overrideComparator()` does not provide access to signature. I did modify `overrideComparator` class to do this. But this was creating conflicts in the existing in code related to override checks for modifiers.

Reply:
Sorry for my misunderstanding, but why do you need access to the signature?

Reply:
The original idea was to check if two modifiers with same name have the same signature. But I realize now that this can be avoided, since modifiers with same name are not allowed to have different signatures.

I'll try to rewrite this.

Reply:
`struct OverrideComparator` doesn't have `operator ==`, so this function has to be here one way or the other. A change could be to just have this as a lambda inside `checkAbstractDefinitions.` 

(A hack would be the following: do two calls to the < operator, a < b and b < a and check if both false; but a very ugly hack.)

Also the function `ModifierType::hasEqualParameterTypes()` can be removed. What do you think @chriseth

Reply:
Well, modifier signature also is needed. I also don't know if we are talking about the same thing. Maybe we can discuss this in a call?

Reply:
I've made some changes in https://github.com/ethereum/solidity/pull/8659/commits/87ba18d1340a245627ab04f28ba515267f16e301

If this isn't what you meant, maybe we should discuss this on a call.